pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +711,131 @@ type ExternalNodeList struct {
 
 	Items []ExternalNode `json:"items,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []SupportBundle `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundle struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of SupportBundle.
+	Spec BundleSpec `json:"spec"`
+	// Most recently observed status of the SupportBundle.
+	Status BundleStatus `json:"status"`
+}
+
+type BundleSpec struct {
+	Nodes         BundleNodes               `json:"nodes,omitempty"`
+	ExternalNodes []BundleExternalNode      `json:"externalNodes,omitempty"`
+	Timeout       int32                     `json:"timeout,omitempty"`
+	Duration      time.Duration             `json:"duration,omitempty"`
```

Comment:
Feels that we should give some comments to the fields of BundleSpec, especially for Timeout and Duration.
And what the format for duration is expected here, like 1d, 12h?

Reply:
Added.